src/QsCompiler/CommandLineTool/CompilationTracker.cs

```diff
@@ -264,6 +263,19 @@ private static void CompilationEventEndHandler(string? parentTaskName, string ta
             task.Stop();
         }
 
+        // Public methods.
+
+        /// <summary>
+        /// Clears tracked data.
+        /// </summary>
+        public static void ClearData()
+        {
+            lock (GlobalLock)
+            {
+                CompilationTasks.Clear();
+            }
+        }
```

Comment:
It is feasible and it would be better than the static class the we currently have. I would like to get the feature branch merged into main soon since some of the functionality in it is needed to inform the next performance improvements to work on, so I created issue #726 to make sure we change the implementation. I will try to work on it in the next few weeks :)